ApiDoctor.Publishing/CSDL/csdlwriter.cs

```diff
@@ -1157,7 +1291,141 @@ private void BuildEntityContainer(EntityFramework edmx, string[] baseUrlsToRemov
             defaultSchema.EntityContainers.Add(container);
         }
 
-        private Dictionary<string, MethodCollection> cachedUniqueRequestPaths { get; set; }
+        private void AddSourceMethods(EntityFramework edmx, string[] baseUrlsToRemove, IssueLogger issues)
+        {
+            Dictionary<string, MethodCollection> uniqueRequestPaths = GetUniqueRequestPaths(baseUrlsToRemove, issues);
+            var resourcePaths = uniqueRequestPaths.Keys.OrderBy(x => x).ToArray();
+
+            foreach (var path in resourcePaths)
+            {
+                try
+                {
+                    var methodCollection = uniqueRequestPaths[path];
+                    if (EntitySetPathRegEx.IsMatch(path))
+                    {
+                        var name = EntitySetPathRegEx.Match(path).Groups[1].Value;
+                        var currentEntitySet = edmx.DataServices.Schemas.SelectMany(c => c.EntityContainers)
+                            .SelectMany(c => c.EntitySets)
+                            .FirstOrDefault(c => c.Name == name);
+                        if (currentEntitySet != null)
+                        {
+                            if (currentEntitySet.SourceMethods != null)
+                            {
+                                var sourceMethods = (MethodCollection)currentEntitySet.SourceMethods;
+                                foreach (var sourceMethod in sourceMethods)
+                                {
+                                    sourceMethod.SplitRequestUrl(out var relativePath, out var queryString,
+                                        out var httpMethod);
+
+                                    foreach (var currentSourceMethod in methodCollection)
+                                    {
+                                        currentSourceMethod.SplitRequestUrl(out var secondRelativePath,
+                                            out var secondQueryString, out var secondMethod);
+                                        if (relativePath != secondRelativePath &&
+                                            queryString != secondQueryString &&
+                                            httpMethod != secondMethod)
+                                        {
+                                            sourceMethods.Add(currentSourceMethod);
+                                        }
+                                    }
+                                }
+                            }
+                            else
+                            {
+                                currentEntitySet.SourceMethods = methodCollection;
+                            }
+                        }
+                    }
+                    else if (SingletonPathRegEx.IsMatch(path))
+                    {
+                        var allSections = FullSingletonPathRegEx.Matches(path);
+                        foreach (var section in allSections)
```

Comment:
Hey @georgend, I'm guessing this PR is not ready, but I'm just taking a peek. Let me know when it's ready. Also it would be cool if we can add a description to the change in the API Doctor versioning wiki.
An example would be nice.